### ISSUE TITLE
Cannot debug code in vscode

### ISSUE THREAD

**Issue by YU5Z8X2CvH1fv4ep** [oracle/truffleruby#2271]
i cant debug my code in vscode.

the text is my launch.json file.
```json
{
    "version": "0.2.0",
    "configurations": [
        {
            "type": "graalvm",
            "request": "launch",
            "name": "Launch Ruby App",
            "outputCapture": "std",
            "protocol": "debugAdapter",
            "runtimeExecutable": "ruby",
            "program": "${workspaceFolder}/App.rb"
        }
    ]
}

```
I have a test script : App.rb
```ruby
def method_name
      a="123"

      print a
end
method_name
```
When i start debugging,will create a new tab,but not stop at breakpoint.

![screenscast_20210224_161302](https://user-images.githubusercontent.com/57204004/108968062-41cf5200-76bb-11eb-895c-565db5581b07.gif)

please help me , thanks!

**Comment by norswap** [comment]
I've been able to reproduce this issue. I have it somehow even worse, as I don't even get the output (`123`) in the console somehow.

Another issue I noticed is that mixing "function breakpoint" (where you type the name of the function in the breakpoint list) and "dot breakpoints" (added in the editor) causes the script to either not run, or to not even stop at the start.

(Function breakpoints also do not work on their own, just like dot breakpoints.)

**Comment by norswap** [comment]
I've filed an issue upstream to the tooling team, I'm waiting on word from there.

**Comment by YU5Z8X2CvH1fv4ep** [comment]
> 我已经向上游工具团队提交了一个问题，我正在等待那里的消息。



> I've filed an issue upstream to the tooling team, I'm waiting on word from there.

thanks

**Comment by entlicher** [comment]
The breakpoint issue was resolved in GraalVM 21.3.0. The path issue is visible on the video - there are two sources named App.rb opened in the editor.

I've verified that with GraalVM 21.3.0 and the latest GraalVM extension 0.5.13 Ruby debugging works as expected.

The output issue needs to be fixed in the Ruby language implementation and there is an internal issue for it.

**Comment by entlicher** [comment]
FYI, you may use `--polyglot-stdio` option as a workaround to get output to the Debug Console.
